components/encoder/fake/encoder.go

```diff
@@ -44,8 +45,8 @@ func NewEncoder(
 	if err := e.Reconfigure(ctx, nil, cfg); err != nil {
 		return nil, err
 	}
+	e.workers = rdkutils.NewStoppableWorkers(e.start)
 
-	e.start(ctx)
```

Comment:
Dang, I think this just fixed a bug! `ctx` was an argument to this function, and would likely be cancelled by the end of reconfiguration. By switching to a `StoppableWorkers`, the context used in the `start` function won't be cancelled until someone calls `Close` on the component. Thanks!

Reply:
heh, the two bugs canceled each other out: our background goroutine would shut down unexpectedly early, and then when we don't shut it down during `Close`, we don't leak goroutines. Thanks for fixing both these issues! :joy:

Reply:
Agreed -- When I saw these issues, `StoppableWorkers` is the first thing that came to mind. Thanks for making it!